Integrations/integration-ProofpointTAP.yml

```diff
@@ -0,0 +1,280 @@
+description: Proofpoint's Targeted Attack Protection (TAP) helps protect against and
+  provide additional visibility into phishing and other malicious email attacks.
+configuration:
+- display: Credentials
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: 'A string specifying which threat type will be returned in the data. If
+    no value is specified, all threat types are returned. The following values are
+    accepted: url, attachment, messageText'
+  name: threatType
+  defaultvalue: ""
+  type: 0
+  required: false
+- display: 'A string specifying which threat statuses will be returned in the data.
+    If no value is specified, active and cleared threats are returned. The following
+    values are accepted: active, cleared, falsePositive'
+  name: threatStatus
+  defaultvalue: ""
+  type: 0
+  required: false
+- display: Fetch incidents
+  name: isFetch
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Incident type
+  name: incidentType
+  defaultvalue: ""
+  type: 13
+  required: false
+- display: Server URL (TAP API URL)
+  name: url
+  defaultvalue: https://tap-api-v2.proofpoint.com
+  type: 0
+  required: true
+- display: API Version (v1, v2)
+  name: apiVersion
+  defaultvalue: v2
+  type: 0
+  required: true
+- display: Do not validate server certificate (insecure)
+  name: insecure
+  defaultvalue: "true"
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: "false"
+  type: 8
+  required: false
+script:
+  script: |-
+    var MINUTE_IN_MILLIS = 60 * 1000;
+    var HOUR_IN_MILLIS = 60 * MINUTE_IN_MILLIS;
+
+    function fixUrl(url) {
+        if (url.endsWith('/')) {
+            return url.slice(0, -1);
+        }
+
+        return url;
+    }
+    var SERVER_URL = fixUrl(params.url) + '/' + params.apiVersion + '/siem';
+
+
+    function httpGet(path) {
+        var requestUrl = SERVER_URL + path;
+
+        var res = http(
+            requestUrl,
+            {
+                Method: 'GET',
+                Username: params.credentials.identifier,
+                Password: params.credentials.password
+            },
+            params.insecure,
+            params.proxy
+        );
+
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nRequest URL: ' + requestUrl +'\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+
+        return JSON.parse(res.Body);
+    }
+
+    function getEvents() {
+        var events = getEventsRequest(args.interval, args.sinceTime, args.threatType, args.threatStatus);
+
+        return {
```

Comment:
Why are there no HR and EC returned?

Reply:
This integration was intended to only fetch incidents.